### ISSUE TITLE
some nodes will fire out errors if set httpAdminRoot:false

### ISSUE THREAD

**Issue by minzojian** [node-red/node-red#2166]
### What are the steps to reproduce?
try to load an exisit simple flow,with a inject and a debug node inside,and set the httpAdminRoot=false
```
var flows=[{"id":"dc77e143.33e97","type":"tab","label":"flow1","disabled":false,"info":""},{"id":"7bd58677.985428","type":"debug","z":"dc77e143.33e97","name":"","active":true,"tosidebar":true,"console":true,"tostatus":false,"complete":"true","x":476,"y":289,"wires":[]},{"id":"32ec7bb.86cd484","type":"inject","z":"dc77e143.33e97","name":"","topic":"","payload":"hello world","payloadType":"str","repeat":"1","crontab":"","once":true,"onceDelay":0.1,"x":237,"y":254,"wires":[["7bd58677.985428"]]}];
var r=require('node-red');
r.init(null,{httpAdminRoot:false})
r.start().then(()=>{
        r.runtime.flows.setFlows({flows:{flows}});
    })
```
### What happens?
got some error,and 
```
18 May 15:57:13 - [warn] ------------------------------------------------------
18 May 16:04:01 - [warn] [node-red/inject] Error: Route.post() requires a callback function but got a [object Undefined] (line:113)
18 May 16:04:10 - [warn] [node-red/debug] Error: Route.post() requires a callback function but got a [object Undefined] (line:151)
18 May 16:04:10 - [warn] [node-red/rpi-gpio] Error: Route.get() requires a callback function but got a [object Undefined] (line:364)
18 May 16:04:10 - [warn] [node-red/udp] Error: Route.get() requires a callback function but got a [object Undefined] (line:133)
18 May 16:04:10 - [warn] ------------------------------------------------------
```
later
```
18 May 16:04:10 - [info] Waiting for missing types to be registered:
18 May 16:04:10 - [info]  - debug
18 May 16:04:10 - [info]  - inject
```

### What do you expect to happen?
i found the code in ``@node-red\nodes\core\core\20-inject.js`` line 113, it says
```
RED.httpAdmin.post("/inject/:id", RED.auth.needsPermission("inject.write"), function(req,res) {
...
```
however ``RED.auth.needsPermission("inject.write")`` return undefined, then error occured
all the four nodes inside the warning looks have same problem 

### Please tell us about your environment:

- [ ] Node-RED version: v0.20.5
- [ ] node.js version: v10.14.2
- [ ] npm version:
- [ ] Platform/OS:
- [ ] Browser:

**Comment by minzojian** [comment]
added a PR with modify the default auth.needPermission function in
``packages\node_modules\@node-red\runtime\lib\index.js``
from 
```
needsPermission: function() {}
```
to
```
needsPermission: function() {return function(req,res,next) {next()}}
```
https://github.com/node-red/node-red/pull/2167

**Comment by knolleary** [comment]
Fixed in master and dev branches via #2167
